instrumentation/powerjob-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/powerjob/v4_0/PowerJobProcessRequest.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.powerjob.v4_0;
+
+import static io.opentelemetry.javaagent.instrumentation.powerjob.v4_0.PowerJobConstants.BASIC_PROCESSOR;
+import static io.opentelemetry.javaagent.instrumentation.powerjob.v4_0.PowerJobConstants.BROADCAST_PROCESSOR;
+import static io.opentelemetry.javaagent.instrumentation.powerjob.v4_0.PowerJobConstants.DYNAMIC_DATASOURCE_SQL_PROCESSOR;
+import static io.opentelemetry.javaagent.instrumentation.powerjob.v4_0.PowerJobConstants.FILE_CLEANUP_PROCESSOR;
+import static io.opentelemetry.javaagent.instrumentation.powerjob.v4_0.PowerJobConstants.HTTP_PROCESSOR;
+import static io.opentelemetry.javaagent.instrumentation.powerjob.v4_0.PowerJobConstants.MAP_PROCESSOR;
+import static io.opentelemetry.javaagent.instrumentation.powerjob.v4_0.PowerJobConstants.MAP_REDUCE_PROCESSOR;
+import static io.opentelemetry.javaagent.instrumentation.powerjob.v4_0.PowerJobConstants.PYTHON_PROCESSOR;
+import static io.opentelemetry.javaagent.instrumentation.powerjob.v4_0.PowerJobConstants.SHELL_PROCESSOR;
+import static io.opentelemetry.javaagent.instrumentation.powerjob.v4_0.PowerJobConstants.SPRING_DATASOURCE_SQL_PROCESSOR;
+
+import tech.powerjob.official.processors.impl.FileCleanupProcessor;
+import tech.powerjob.official.processors.impl.HttpProcessor;
+import tech.powerjob.official.processors.impl.script.PythonProcessor;
+import tech.powerjob.official.processors.impl.script.ShellProcessor;
+import tech.powerjob.official.processors.impl.sql.DynamicDatasourceSqlProcessor;
+import tech.powerjob.official.processors.impl.sql.SpringDatasourceSqlProcessor;
+import tech.powerjob.worker.core.processor.sdk.BasicProcessor;
+import tech.powerjob.worker.core.processor.sdk.BroadcastProcessor;
+import tech.powerjob.worker.core.processor.sdk.MapProcessor;
+import tech.powerjob.worker.core.processor.sdk.MapReduceProcessor;
+
+public final class PowerJobProcessRequest {
+  private String methodName;
+  private final Long jobId;
+  private String jobType;
+  private Class<?> declaringClass;
+
+  private boolean failed;
+
+  private String jobParams;
+  private String instanceParams;
+
+  private PowerJobProcessRequest(Long jobId) {
+    this.jobId = jobId;
+  }
+
+  public static PowerJobProcessRequest createRequest(
+      Long jobId, BasicProcessor handler, String methodName) {
+    PowerJobProcessRequest request = new PowerJobProcessRequest(jobId);
+    request.methodName = methodName;
+    request.declaringClass = handler.getClass();
+    request.jobType = BASIC_PROCESSOR;
```

Comment:
I think you could make it shorter and delete the constants with something like.

```java
private static final List<Class<?>> knownProcessors = Arrays.asList(BroadcastProcessor.class, MapProcessor.class);

for (Class<?> processorClass : knownProcessors) {
  if (processorClass.isInstance(handler)) {
    request.jobType = processorClass.getSimpleName();
    break;
  }
}
```

Reply:
Done